profile-specific-challenges/frontend/components/idioms.jsx

```diff
@@ -346,15 +378,19 @@ export function UnoptimizableRenderingStructure(altRecords) {
     loadRecords();
   }, []); 
 
+  // you can either do this or use the React.memo HOC by creating another component
+  const memoizedAltRecords = useMemo(() => (
+    <ul>
+      {altRecords.map(rec => <li key={rec.id}>{rec.id}</li>)}
+    </ul>
+  ), [altRecords]);
```

Comment:
Just like at the beginning of this file (components as variables) I don't like jsx named used this way. 
Also this useMemo is also a memory tradeoff for all possible alt records. Probably not good. I think it should just be a different react component.